src/lifecycle/properties.js

```diff
@@ -77,7 +77,7 @@ function property (elem, name, prop) {
 
       if (isBoolean && internalValue) {
         this.setAttribute(prop.attr, '');
-      } else if (internalValue == null || isBoolean && !internalValue) {
+      } else if (value == null || isBoolean && !internalValue) {
```

Comment:
In my opinion if the value is `undefined` then I think that we should skip calling `prop.type()` on it altogether. I can't imagine anyone would want to handle `undefined` as a valid value for a component's property (perhaps `null` would be appropriate as it often intentionally denotes "not applicable", "empty", etc. for example in `Element#querySelector(".no-elements-with-this-class") === null`, `Element#getAttribute("no-such-attribute") === null`).